### ISSUE TITLE
Box inputs

### ISSUE THREAD

**Issue by biendltb** [ryouchinsa/sam-cpp-macos#14]
Hi,

Thank you for the great work.

When I download the exported models of SAM 2.1 from HuggingFace repo that you provided, I don't see the box prompt in the inputs of the model. Do you know why?

![Image](https://github.com/user-attachments/assets/859dc683-c23c-4e56-9281-c7e149f670fa)

**Comment by biendltb** [comment]
Found the answer. In SAM 2.1, they [concatenate the box coords with points coords](https://github.com/SysCV/sam-hq/blob/afb90d71602048d0e74d4a0bbf6b8b40e3e27f26/sam-hq2/sam2/sam2_image_predictor.py#L403) and label them with 2 and 3 for the first and second point of the box respectively.

```py
        # Embed prompts
        if boxes is not None:
            box_coords = boxes.reshape(-1, 2, 2)
            box_labels = torch.tensor([[2, 3]], dtype=torch.int, device=boxes.device)
            box_labels = box_labels.repeat(boxes.size(0), 1)
            # we merge "boxes" and "points" into a single "concat_points" input (where
            # boxes are added at the beginning) to sam_prompt_encoder
            if concat_points is not None:
                concat_coords = torch.cat([box_coords, concat_points[0]], dim=1)
                concat_labels = torch.cat([box_labels, concat_points[1]], dim=1)
                concat_points = (concat_coords, concat_labels)
            else:
                concat_points = (box_coords, box_labels)
```
